### PR TITLE
New package: lean-community-3.32.1

### PR DESCRIPTION
**Pull request by Eloitor**
#### General
- [x] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

 
#### Does it build and run successfully? 
(Please choose at least one native build and, if supported, at least one cross build. More are better.)
- [x] I built this PR locally for my native architecture, (ARCH-LIBC)
- [ ] I built this PR locally for these architectures (if supported. mark crossbuilds):
  - [ ] aarch64-musl
  - [ ] armv7l
  - [ ] armv6l-musl

### CODE REVIEW

File: srcpkgs/lean-community/template [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+pkgname=lean-community
+version=3.32.1
+revision=1
+wrksrc="lean-$version"
+build_wrksrc=src
+build_style=cmake
+make_cmd=ninja
+hostmakedepends="cmake ninja gmp"
+makedepends="gmp-devel"
+depends="gmp"
```
Comment (tornaria) [inline srcpkgs/lean-community/template:11]: The runtime dependency on `gmp`  is not needed, it should be automatic based on sonames.

Comment (Eloitor) [reply srcpkgs/lean-community/template]: Thanks, I removed it.

**Comment by tornaria** [comment]
It seems the build tries to run `lean` itself (to compile some lean files?) meaning that it's not obvious how to cross-build. Unless there's another way, you should add `nocross=yes` to template (right after `checksum`).

I compiled the template in x86_64, x86_64-musl, and i686.

All 1414 tests pass, installs and works in x86_64. Same for -musl (tested in a chroot).

I had 1 test failure in i686:
```
[...]
99% tests passed, 1 tests failed out of 1414

Total Test time (real) = 391.49 sec

The following tests FAILED:
	1340 - leanittest_complete_trailing_period.lean (Failed)
Errors while running CTest
FAILED: CMakeFiles/test.util
[...]
```

@ericonr can you approve the workflow for this PR?

**Comment by tornaria** [comment]
Here's the complete error for test #1340 on i686, in case you can figure out what's wrong.
```
1340/1414 Test #1340: leanittest_complete_trailing_period.lean ....................***Failed    0.03 sec
--- complete_trailing_period.lean.expected.out	2021-08-12 10:56:43.000000000 +0000
+++ complete_trailing_period.lean.produced.out	2021-08-19 22:20:09.954646641 +0000
@@ -1,2 +1,2 @@
 {"message":"file invalidated","response":"ok","seq_num":0}
-{"completions":[{"source":,"text":"foo.rec","type":"Π (motive : foo → Sort l) (n : foo), motive n"},{"source":{"column":10,"line":3},"text":"foo.rec_on","type":"Π (motive : foo → Sort l) (n : foo), motive n"},{"source":{"column":10,"line":3},"text":"foo","type":"Type"}],"prefix":"foo.","response":"ok","seq_num":6}
+{"completions":[],"prefix":"foo.","response":"ok","seq_num":6}
ERROR: file complete_trailing_period.lean.produced.out does not match complete_trailing_period.lean.expected.out
```

**Comment by Eloitor** [comment]
> It seems the build tries to run `lean` itself (to compile some lean files?) meaning that it's not obvious how to cross-build. Unless there's another way, you should add `nocross=yes` to template (right after `checksum`).
> 

Done. When I know more about compiling and void I'll try to figure out if it's possible to cross-compile it.

**Comment by tornaria** [comment]
Could you try to figure out what's the issue with the failing test in i686?

Here's how I easily test i686. In my void-packages I have a script named `xbps-src-i686` as follows:
```sh
#! /bin/sh

if [ "$1" = "binary-bootstrap" ] ; then
    set -- binary-bootstrap i686
fi

./xbps-src -m masterdir.i686 "$@"
```
Then you just use `./xbps-src-i686` instead of `./xbps-src`  (remember to run `./xbps-src-i686 binary-bootstrap` once to begin).

You can do the same for musl by replacing `i686` with `x86_64-musl`.

**Comment by Eloitor** [comment]
I get a lot of warnings like
```
=> WARNING: lean-community-3.32.1_1: removed empty dir: /usr/include/lean_ext/tests/util/numerics
=> WARNING: lean-community-3.32.1_1: removed empty dir: /usr/include/lean_ext/tests/util
=> WARNING: lean-community-3.32.1_1: removed empty dir: /usr/include/lean_ext/tests/shell
```

**Comment by tornaria** [comment]
Those warnings seem harmless. The only issue I see here is to figure out what's going on with the test on i686, and either fix it or disable it.

**Comment by tornaria** [comment]
The test failure on i686 still happens on current version of lean (3.39.2), I've reported it upstream.

**Pull request closed without merging**